### PR TITLE
jwm-settings-manager: init at 2018-10-19

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

Add [jwm-settings-manager](https://github.com/Israel-D/jwm-settings-manager).

> A full configuration manager for JWM, enabling the average user the ability to customize JWM to their liking. Built using FLTK and pugixml, this is a fast and light application to configure one of the fastest lightest Window Managers.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/jwm/jwm-settings-manager.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    pkgconfig
+    gettext
+  ];
+
+  buildInputs = [
+    libXpm
+    libGL
+    fltk
+    hicolor-icon-theme
+    which
```
Comment (xeji) [inline pkgs/applications/window-managers/jwm/jwm-settings-manager.nix]: Is `which` really used at runtime? If not, it belongs in `nativeBuildInputs`.

File: pkgs/applications/window-managers/jwm/jwm-settings-manager.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      --replace 'CMAKE_INSTALL_PREFIX "/usr"' "CMAKE_INSTALL_PREFIX $out"
+    substituteInPlace data/CMakeLists.txt \
+      --replace 'DESTINATION usr/share' "DESTINATION share"
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A full configuration manager for JWM";
+    homepage = https://joewing.net/projects/jwm;
+    license = licenses.gpl3;
+    platforms = platforms.unix;
```
Comment (xeji) [inline pkgs/applications/window-managers/jwm/jwm-settings-manager.nix]: Is this supposed to work on MacOS? Please either fix the Darwin build or restrict platforms to linux.

Comment (7c6f434c) [reply pkgs/applications/window-managers/jwm/jwm-settings-manager.nix]: As JWM is an X11 WM, I guess macOS support for editing its configuration is not too useful, even if possible.

File: pkgs/applications/window-managers/jwm/jwm-settings-manager.nix (hunk 1 above)
Comment (romildo) [reply pkgs/applications/window-managers/jwm/jwm-settings-manager.nix]: Yes, `which` is used at runtime to locate optional executables.

File: pkgs/applications/window-managers/jwm/jwm-settings-manager.nix (hunk 2 above)
Comment (romildo) [reply pkgs/applications/window-managers/jwm/jwm-settings-manager.nix]: Ok, changing it to linux.

**Pull request merged**